libcloudforensics/providers/gcp/forensics.py

```diff
@@ -30,19 +30,19 @@
 def CreateDiskCopy(
     src_proj: str,
     dst_proj: str,
-    instance_name: str,
     zone: str,
+    instance_name: Optional[str] = None,
```

Comment:
why not ( : str = '' ) similar to other places in the code, I am not sure but I believe you are removing None and using empty objects to avoid explicit type comments or ignore statements

Reply:
See #171 for context

Reply:
A bit more info on this one, as I realise my previous answer doesn't really fit your comment: prior to this PR `instance_name` was a required parameter, which makes little sense when you want to use `disk_name` which is optional (especially on the CLI, where it's cumbersome having to use the instance name + the disk name if you already know the latter to make your copy). To make it consistent with what we have with AWS (and soon Azure) and make the CLI a little more user friendly, I switched `instance_name` to optional. Then, why initialise it to `None` and not to `''` is just a personal preference on my side. If we want to assign it `''` then I would opt to create a separate issue to track this and go over the entire codebase to make it consistent with other optional parameters of type `string`.
Thoughts @tomchop @hacktobeer @sa3eed3ed ?

Reply:
My 2 cents is that it should be Optional if from a functional perspective the parameter *is* optional. So the typing should reflect the actual status of a parameter from the user perspective.

Setting a str to None or '' I am indifferent to. Afaik code flows (eg 'if str:')treat a str that is None or '' the same. As long as we use it consistently through the codebase.

Reply:
I agree that it is indifferent, I just realised that None is replaced with a '' in other places in the code as well and was wondering if there is a specific reason, I vote for consistency and I my subjective preference is None

Reply:
#189

Reply:
FWIW I think it's better to use `None` than empty strings. Means that string functions like `len` or `split` will fail and bugs will be easier to debug.